pulpcore/cli/common/generic.py

```diff
@@ -852,7 +849,7 @@ def callback(pulp_ctx: PulpContext, entity_ctx: PulpEntityContext, **kwargs: Any
         """
         Create a {entity}.
         """
-        body: EntityDefinition = entity_ctx.preprocess_body(kwargs)
+        body: EntityDefinition = entity_ctx.preprocess_entity(kwargs, partial=False)
```

Comment:
I would like to move the call for `preprocess_entity` into the `create` and `update` calls, but it's not idempotent, and we may run into trouble with external plugins. Suggestions?

Reply:
- Cache the results of `preprocess_entity`
- Add a field to result of `preprocess_entity` that `create` and `update` would remove and know not to run it a second time
- Change the function signature of `create` and `update` to take in an unprocessed body
- In `create` & `update` run `preprocess_entity` on a copy of `body` and then **magically** chose the correct one to use.

Reply:
I went with a slight modification of your second suggestion: A subclass of dict.